### PR TITLE
illustrate a message

### PR DESCRIPTION
**Pull request by LotuxPunk**

### CODE REVIEW

File: lib/webhook.js [hunk 1]
```diff
@@ -104,9 +104,9 @@ class Webhook {
         return this.sendPayload(payload);
     }
 
-    custom(name, message, title, color){
+    custom(name, message, title, color, imageUrl){
         let payload;
-        if(color){
+        if(color && imageUrl){
```
Comment (JoeBanks13) [inline lib/webhook.js]: This is a great addition, but I don't really like all these if loops, is it possible to reduce all this into one `payload = ...` and then
```js
if (color) {
    // do stuff to add color to payload
}
```
and 
```js
if (imageUrl) {
    // add image URL to payload
}
```

File: lib/webhook.js [hunk 2]
```diff
@@ -117,25 +117,71 @@ class Webhook {
...
+                        "ts": new Date() / 1000
+                    }]
+                }
             }
         }
 
         return this.sendPayload(payload);
     }
+
+    illustrated(name, message, title, imageUrl) {
```
Comment (JoeBanks13) [inline lib/webhook.js]: What is the use of this illustrated function, does it just add an image, if so, I think that `image` or something would be a more applicable name, I'm not sure if people would understand what the illustrated function would do.

File: README.md [hunk 3]
```diff
@@ -35,7 +35,12 @@ Hook.success("WEBHOOK NAME","Yay we did something right")
 
 To send a message with a custom format:
 ```js
-Hook.custom("WEBHOOK NAME","MESSAGE CONTENT","MESSAGE TITLE","MESSAGE COLOUR (optional)")
+Hook.custom("WEBHOOK NAME","MESSAGE CONTENT","MESSAGE TITLE","MESSAGE COLOUR (optional)", "IMAGE URL (optionnal)")
+```
+
+To send a message with a custom format and an image:
```
Comment (JoeBanks13) [inline README.md]: This section pushes the bit about the `custom` functions `colour` argument to the bottom. This should be underneath the message about color.

Comment (JoeBanks13) [reply README.md]: As well as that, apologies for my spelling of colour, I'm chopping and changing between the English and American spelling :P

Comment (JoeBanks13) [reply README.md]: Also just noticed a typo in the word "optional" while looking at this PR later in the day.

**Comment by LotuxPunk** [comment]
Hello ! 
I apologize for my rudimentary English, I am a French-speaking Belgian. I have tried to understand the changes requested, and apart from my "payload.concat()", I am almost sure of myself.
I'll check this function later. ;)

**Comment by JoeBanks13** [comment]
Just seen changes, all looks good apart from the README.

The note about how color has to be hexadecimal should be with the bit which talks about the custom function.

**Comment by LotuxPunk** [comment]
And now ? 😃

**Comment by JoeBanks13** [comment]
Looks good to me, just need to update this with the new stuff on `master`. Then I'll test it out.

**Comment by JoeBanks13** [comment]
I'm currently testing this and neither the image in `custom` or `image` is working? Am I calling this correctly:
```js
> x.custom("test", "test", "test", null, "https://josephbanks.me/background1.jpg")
> x.image("test", "test", "test", "https://josephbanks.me/background1.jpg")
```

**Comment by LotuxPunk** [comment]
Yes

**Comment by JoeBanks13** [comment]
Not appearing in my tests

**Comment by LotuxPunk** [comment]
I've forgot a 's' to "embeds" https://gist.github.com/Birdie0/78ee79402a4301b1faf412ab5f1cdcf9

**Comment by LotuxPunk** [comment]
It's normally work now

**Comment by LotuxPunk** [comment]
So what I did didn't work out?
If so, do you have the API documentation used?

**Comment by LotuxPunk** [comment]
https://api.slack.com/docs/message-attachments

**Comment by LotuxPunk** [comment]
Damn :D https://birdie0.github.io/discord-webhooks-guide/other/slack_formatting.html

**Comment by JoeBanks13** [comment]
That's awesome! Thank you for your contribution. I'll merge this and then push the new version to NPM!

**Comment by LotuxPunk** [comment]
Thank you! I will pull some other cool features in the future. Thank you for this project. 😉

**Pull request merged**